src/Nethermind/Nethermind.Synchronization/FastBlocks/SyncStatusList.cs

```diff
@@ -77,6 +79,80 @@ public void GetInfosForBatch(BlockInfo?[] blockInfos)
             }
         }
 
+        /// <summary>
+        /// Try get block infos of size `batchSize`.
+        /// </summary>
+        /// <param name="batchSize"></param>
+        /// <param name="blockExist"></param>
+        /// <param name="infos"></param>
+        /// <returns></returns>
+        public bool TryGetInfosForBatch(int batchSize, Func<BlockInfo, bool> blockExist, out BlockInfo?[] infos)
+        {
+            BlockInfo?[] outputArray = new BlockInfo?[batchSize];
+            BlockInfo?[] workingArray = new BlockInfo?[batchSize];
```

Comment:
Wonder if we could pool memory here. Probably yes ad `BodiesSyncBatch` and `ReceiptsSyncBatch` already do that for response, so why not for `Infos`?

Reply:
Pooled workingArray.